src/main/java/com/gluonhq/RunAgentMojo.java

```diff
@@ -0,0 +1,121 @@
+import org.apache.maven.plugins.annotations.LifecyclePhase;
+import org.apache.maven.plugins.annotations.Mojo;
+import org.apache.maven.plugins.annotations.ResolutionScope;
+import org.apache.maven.shared.invoker.*;
+
+import java.io.*;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.stream.Collectors;
+
+@Mojo(name = "run", requiresDependencyResolution = ResolutionScope.RUNTIME)
+@Execute(phase = LifecyclePhase.PROCESS_CLASSES)
+public class RunAgentMojo extends NativeBaseMojo {
+
+    private static final String POM_XML = "pom.xml";
+    private static final String RUN_POM_XML = "runPom.xml";
+
+    @Override
+    public void execute() throws MojoExecutionException {
+        try {
+            createSubstrateDispatcher();
+        } catch (Exception e) {
+            e.printStackTrace();
+            throw new MojoExecutionException("Error creating Substrate Dispatcher", e);
+        }
+
+        final InvocationRequest invocationRequest = new DefaultInvocationRequest();
+        invocationRequest.setProfiles(project.getActiveProfiles().stream()
+                .map(Profile::getId)
+                .collect(Collectors.toList()));
+        invocationRequest.setProperties(session.getRequest().getUserProperties());
+
+        // 1. Read pom
+        File pomFile = new File(POM_XML);
+        // 2. Create temporary pom file
+        File runPomFile = new File(RUN_POM_XML);
+        try (InputStream is = new FileInputStream(POM_XML)) {
+            // 3. Create model from current pom
+            Model model = new MavenXpp3Reader().read(is);
+
+            model.getBuild().getPlugins().stream()
+                    .filter(p -> p.getGroupId().equalsIgnoreCase("org.openjfx") &&
+                            p.getArtifactId().equalsIgnoreCase("javafx-maven-plugin"))
+                    .findFirst()
+                    .orElseThrow(() -> new MojoExecutionException("No JavaFX plugin found"));
+            
+            // 4. Check for Attach Dependencies and if Desktop is supported, add the classifier
+            model.getDependencies().stream()
+                    .filter(p -> p.getGroupId().equalsIgnoreCase("com.gluonhq.attach") &&
+                            Arrays.stream(AttachService.values()).filter(AttachService::isDesktopSupported).anyMatch(attach -> attach.getServiceName().equalsIgnoreCase(p.getArtifactId())))
+                    .forEach(p -> p.setClassifier("desktop"));
+
+            // 5. Serialize new pom
+            try (OutputStream os = new FileOutputStream(runPomFile)) {
+                new MavenXpp3Writer().write(os, model);
+            }
+        } catch (Exception e) {
+            if (runPomFile.exists()) {
+                runPomFile.delete();
+            }
+            throw new MojoExecutionException("Error reading pom", e);
+        }
+
+        invocationRequest.setPomFile(runPomFile);
+        invocationRequest.setGoals(Collections.singletonList("javafx:run"));
```

Comment:
We deliberately ignore any active profile here, so `mvn -Pandroid client:run` is the same as `mvn client:run`, since we just run `javafx:run` on the host machine.
I'm not sure we should show a warning though: check if the active profile is not desktop, then print something like "the active profile $profile will be ignored to run javafx:run on the host machine".

Reply:
I don't see an easy way to get the name of the activated profile. Are you aware of a possible solution?

Reply:
See `NativeBuildMojo`.